### ISSUE TITLE
Can you add autocomplete function?

### ISSUE THREAD

**Issue by tolerious** [matiastucci/vue-input-tag#2]
<img width="578" alt="2016-11-25 1 38 16" src="https://cloud.githubusercontent.com/assets/2922120/20615388/77ef0d36-b314-11e6-9e56-23d3c6f9e9ab.png">

such as this, get the prompt data from backend.

**Comment by matiastucci** [comment]
Hi! I still have to finish some stuff and be sure that the componen is stable. Once I finish with that I would like to implement this. Thanks for the suggestion.

**Comment by tolerious** [comment]
THX, I am using your component in my project.

**Comment by HectorMalot** [comment]
Can I suggest to leave this issue open to keep track of the work for autocomplete?

**Comment by unitedworx** [comment]
further to autocomplete it would be nice if validators can talk to a backend. e.g imagine tags being products codes and a validator being able to talk to the backend using an axios/ajax call and return trey or false if the product exists or not.

nice work on this component.

**Comment by pdey** [comment]
Hi. Thanks for your work on this component.
I also wanted an autocomplete functionality but i would like to compose a custom component with this one to achieve the same. However, i need a callback on the event of input change, so that i can use the partial input text to generate a list of autocomplete options. Can i send a pull-request for this change? Should be very simple.

**Comment by matiastucci** [comment]
sure :)

**Comment by richlove1** [comment]
Autocomplete / callback on-change functionality would be super cool 👍

**Comment by EvanZ** [comment]
Basically just need a @keyup functionality for this right? With that as a minimum, we could implement autocomplete in our client relatively straightforward.

**Comment by michaelmano** [comment]
I might make a fork of this to work on this since I require it for a project but don't wish to reinvent the wheel.

Also since its just used for tags I don't think an http call is required since most of the time you are just checking if tags exist. maybe just pass it an array like the pre filled ones then checking if the value exists in array if so suggest it, to start with at least.

**Comment by unitedworx** [comment]
@michaelmano  think of tags being products codes and products being tens of thousands, does not make sense to give it an array of thousands of valid product codes but make much sense to make an Ajax call to validate if the code exists! if the validation is implemented in some sort of function callback then its up to you to simply check an array or make an Ajax call depending on your needs!

**Comment by michaelmano** [comment]
@unitedworx then maybe have either option. An auto complete array for smaller things and an auto complete URL that will post with a denounce function to get an array

**Comment by iamdubx** [comment]
Autocomplete is must have feature, please consider to add it.

**Comment by LorrainInfinity** [comment]
Mmmmh this component is really useful.  Too bad it's not 100% finished, it lacks a few methods you don't seem to be willing to implement I guess I'll have to do it on my own ^^'

**Comment by ianaya89** [comment]
@LorrainInfinity Sometimes is really hard to make enough time to maintain an open source project but PR's are welcome 😉

**Comment by LorrainInfinity** [comment]
@ianaya89 Yes that's true ^^. It doesn't lack much and I think it could be useful enough to a lot of people but I'm trying to use it for a very complex architecture so I'll have to improve it a bit :p

**Comment by matiastucci** [comment]
@LorrainInfinity besides this feature, what else is missing in your opinion?

**Comment by LorrainInfinity** [comment]
It'd be great to be able to detect which element was deleted.  I know you did add a update:tags method (btw you should add it on the home page of the git project so people know about it :p).  But I need to know exactly what element is removed from the array of elements.

On another hand it'd be great to be able to pass an array of objects as tags, and if so, add a "track-by" prop so it tracks based on a specific key.  That's because I need to create an array of strings instead of using my real array of objects.

**Comment by EvanZ** [comment]
@LorrainInfinity You might check out Vuetify which pretty much has all these features in their multi-select. https://vuetifyjs.com/en/components/selects#api

**Comment by LorrainInfinity** [comment]
Mmh yes maybe it could work but what I really need here is a tag creator not a multiselect (even though I saw that there is a "tag" option to allow tag creation, but in the end, it'll still be a select list isn't it?). 

Anyway thank you for the component I'll look into that :)

**Comment by EvanZ** [comment]
You can use it as a multi-select but also add new tags.

**Comment by tomaytotomato** [comment]
Yea this feature will be very useful. 

I am happy to contribute my solution after I've tried hacking it

**Comment by demonodojo** [comment]
I have added this PR to allow implement autocomplete out of the component:
https://github.com/matiastucci/vue-input-tag/pull/63
